cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -108,7 +108,11 @@ class cufile_shim {
 
   ~cufile_shim()
   {
-    if (driver_close != nullptr) driver_close();
+    // Explicit cuFile driver close should not be performed here to avoid segfault. However, in the
+    // absence of driver_close(), cuFile will implicitly do that, which in most cases causes
+    // segfault anyway. TODO: Revisit this conundrum once cuFile is fixed.
+    // https://github.com/rapidsai/cudf/issues/17121
```

Comment:
Please add a reference to the issue to the PR description.

Also, I don't think description is right - we aren't just adding comments.

Reply:
Thanks! Done.